app/components/icons/tokens/IconDefault.tsx

```diff
@@ -1,13 +1,20 @@
+import randomColor from 'randomcolor'
 import * as React from 'react'
+import { useEffect, useState } from 'react'
 import Svg, { Circle, SvgProps, Text } from 'react-native-svg'
 
 export function IconDefault (symbol: string): (props: SvgProps) => JSX.Element {
   return (props: SvgProps): JSX.Element => {
+    const [color, setColor] = useState('')
+    useEffect(() => {
+      const c = randomColor({ luminosity: 'dark', seed: symbol })
+      setColor(c)
+    }, [symbol])
```

Comment:
instead of random, it should be hash-based color derived from the token name and being deterministic

---

And assuming the above is true since your random is deterministic you don't need to `useEffect` and you can just use do a normal function call?

Reply:
Ah yeah good point, I'll remove the `useEffect`. it's already based on the token name (seed option)